common/include/opae/metrics.h

```diff
@@ -0,0 +1,113 @@
+// Copyright(c) 2018, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/**
+* @file metrics.h
+* @brief Functions for Discover/ Enumerates metrics and retrieves values
+*
+*
+*
+*
+*/
+
+#ifndef __FPGA_METRICS_H__
+#define __FPGA_METRICS_H__
+
+#include <opae/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Enumerates number of metrics
+ *
+ * @param[in]  handle     Handle to previously opened fpga resource
+ * @param[inout] num_metrics number of metrics are discovered in
+ *                        fpga resource
+ * @returns FPGA_OK on success. FPGA_NOT_FOUND if the Metrics are not
+ * discovered
+ *
+ */
+fpga_result fpgaGetNumMetrics(fpga_handle handle,
+				uint64_t *num_metrics);
+
+/**
+ * Retrieve metrics information
+ *
+ * @param[in]  handle     Handle to previously opened fpga resource
+ * @param[inout] metric_info pointer to array of metric info struct
+ *                        user allocates metrics info array
+ * @param[inout] num_metrics size of metric info array
+ *
+ * @returns FPGA_OK on success. FPGA_NOT_FOUND if the Metrics are not
```

Comment:
are there any other possible return codes? Is it possible that some code path leads to something returning FPGA_NO_MEMORY?